pkg/helper/job.go

```diff
@@ -54,3 +56,60 @@ func HasJobCompleted(jName string, jNamespace string, client k8sclient.Client) b
 
 	return false
 }
+
+// HasJobImageChanged returns true if the Job's existing image is different from the provided image
+func HasJobImageChanged(jName string, jNamespace string, desiredImage string, client k8sclient.Client) (bool, error) {
+	job := &batchv1.Job{}
+	err := client.Get(context.TODO(), k8sclient.ObjectKey{
+		Namespace: jNamespace,
+		Name:      jName,
+	}, job)
+
+	// Return error if can't get Job
+	if err != nil && !k8serr.IsNotFound(err) {
+		return false, fmt.Errorf("error getting job %s: %w", job.Name, err)
+	}
+
+	// Return false if the Job doesn't exist yet
+	if k8serr.IsNotFound(err) {
+		return false, nil
+	}
+
+	// Return true if the Job is using the old image
+	if job.Spec.Template.Spec.Containers[0].Image != desiredImage {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func DeleteJob(jName string, jNamespace string, client k8sclient.Client) error {
```

Comment:
I'm trying to understand what means Delet Job. Please correct me if I'm mistaken. Job it's somethings that actually initially not exists (if it's not CRON job). Job exists just when it's running (let's say somone is running Job - user, script, maybe operator). Job is running and completing; this is difference from Deployment, that running pod as service. Job - no, it do something and it will be "soon" completed. Both deploy and job are using "pods templates" (spec_, that can be changed. If it's deployment - will be rolling update in this case (automatic). If it's a Job ... - just need let old job  to complete, and will run new job, when needed. Delete Job, as I understand  - means to Cancel some work that in progress currently. Is it something that we want to do? Please correct me if I'm wrong. Thank you

Reply:
@valerymo the Job won't be deleted unless there is an image change which would only happen during upgrades. The chances of two upgrades happening before the Job has time to complete is very remote so I don't think we need to consider it here.